### PR TITLE
feat(service/JWT):#47- JWT implementation

### PR DESCRIPTION
**Pull request by karthikeyanpa12**
## Steps To Implement Feature

JWT

**JWT Implementation**
  
 1) **JSON Web Token (https://jwt.io/)**
         A JSON Web Token (JWT) is an open standard (RFC 7519) that defines a compact and self-contained way for securely transmitting information between parties as a JSON object

 2) **JWT contains 3 parts :**
                Header, Payload, Signature

       **Example Token:** eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1c2VyMSIsImV4cCI6MTU5ODU2MTI5MSwiaWF0IjoxNTk4NTI1MjkxfQ.R4xLFgVxkh1pO--wfjP-kE-Gq0DeFGNk5Fls-PhjEqY
 
  **Header:** Type of algorithm
                                    {"alg": "HS256" }
  **Payload**:  Username and required details 
        	   { "sub": "user1", "exp": 1598561291,  "iat": 1598525291}
  **Signature** 
              HMACSHA256( base64UrlEncode(header) + "." +base64UrlEncode(payload)) 

**Implementation**:
  **User** - User model contains user details
  **User Repository** - **findByUserName** JPA method to get user details
  **User Request** - request method properties
  **UserService** - Custom implementation for **findByUserName**

 **JWT Util:**
           **createToken** method to generate token for valid user
           **validateToken**  method validate the service call based on token

**JWT Filter** - validate all the service call to check the valid token exist using doFilterInternal method
**AuthenticationController** - authenticate service call for initial request

**Dependency(Pom.xml)**
  ```
 <dependency>
     	 <groupId>io.jsonwebtoken</groupId>
      	<artifactId>jjwt</artifactId>
      	<version>0.9.1</version>
    </dependency>
    <dependency>
     	 <groupId>org.springframework.boot</groupId>
     	 <artifactId>spring-boot-starter-security</artifactId>
    </dependency>
    <dependency>
      	<groupId>org.springframework.security</groupId>
      	<artifactId>spring-security-test</artifactId> 
     	 <scope>test</scope>
    </dependency>
```

Authenticate

BoardList

### CODE REVIEW

**Review by naveenvignesh5 — APPROVED** [review]

File: services/src/main/java/io/trello/trelloapp/authentication/JwtUtil.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    public String generateToken(String username) {
+        Map<String, Object> claims = new HashMap<>();
+        return createToken(claims, username);
+    }
+
+    private String createToken(Map<String, Object> claims, String subject) {
+
+        return Jwts.builder().setClaims(claims).setSubject(subject).setIssuedAt(new Date(System.currentTimeMillis()))
+                .setExpiration(new Date(System.currentTimeMillis() + 1000 * 60 * 60 * 10))
```
Comment (ramsunvtech) [inline services/src/main/java/io/trello/trelloapp/authentication/JwtUtil.java:47]: can this calculation be in separate method.

File: services/src/main/java/io/trello/trelloapp/TrelloCloneApp.java [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
 import org.springframework.context.annotation.Bean;
 import org.springframework.web.servlet.config.annotation.CorsRegistry;
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
 
+import javax.annotation.PostConstruct;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
 @SpringBootApplication
```
Comment (ramsunvtech) [inline services/src/main/java/io/trello/trelloapp/TrelloCloneApp.java:18]: do we need another Application. is this separate micro service ?

File: services/src/main/java/io/trello/trelloapp/authentication/UserRequest.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package io.trello.trelloapp.authentication;
+
+public class UserRequest {
```
Comment (ramsunvtech) [inline services/src/main/java/io/trello/trelloapp/authentication/UserRequest.java:3]: may be later, maintain the models for request and response is better

File: services/src/main/java/io/trello/trelloapp/TrelloCloneApp.java [hunk 4]
```diff
@@ -17,7 +25,21 @@ public static void main(String[] args) {
 		SpringApplication.run(TrelloCloneApp.class, args);
 	}
 
-  @Bean
+	@Autowired
+	private UserRepository repository;
+
+	@PostConstruct
+	public void initUsers() {
+		List<User> users = Stream.of(
```
Comment (ramsunvtech) [inline services/src/main/java/io/trello/trelloapp/TrelloCloneApp.java:33]: why are we going with Streams ?

**Review by ramsunvtech — APPROVED** [review]

**Pull request merged**